processor/transformprocessor/internal/common/func_replace_pattern.go

```diff
@@ -0,0 +1,40 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func replacePattern(target GetSetter, regexPattern string, replacement string) (ExprFunc, error) {
+	compiledPattern, err := regexp.Compile(regexPattern)
+	if err != nil {
+		return nil, fmt.Errorf("the regex pattern supplied to replace_pattern is not a valid pattern: %w", err)
+	}
+	return func(ctx TransformContext) interface{} {
+		originalVal := target.Get(ctx)
+		if originalVal == nil {
+			return nil
+		}
+		if originalValStr, ok := originalVal.(string); ok {
+			if compiledPattern.MatchString(originalValStr) {
+				updatedStr := compiledPattern.ReplaceAllLiteralString(originalValStr, replacement)
```

Comment:
is regex2 better? https://github.com/dlclark/regexp2. it provides full regex syntax support. Golang itself does not support lookahead/lookbehind function.

Reply:
Contributing guidelines here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/CONTRIBUTING.md#new-functions

Reply:
I was following the pattern of replace_match implementation.  Should I take out the compile of patterns before each check and just put that in the startup?